### PR TITLE
Prem AI Vectorizer support

### PR DESCRIPTION
**Pull request by Anindyadeep**
This PR adds support for Prem AI Embeddings in DSPy `modules.sentence_vectorizer`

### CODE REVIEW

**Comment by krypticmouse** [comment]
are the changes in dataloader just lint fixes?

File: dsp/modules/sentence_vectorizer.py [hunk 1]
```diff
@@ -247,4 +252,65 @@ def __call__(self, inp_examples: List["Example"]) -> np.ndarray:
         texts_to_vectorize = self._extract_text_from_examples(inp_examples)
         embeddings = self._model.embed(texts_to_vectorize, batch_size=self._batch_size, parallel=self._parallel)
 
-        return np.array([embedding.tolist() for embedding in embeddings], dtype=np.float32)
\ No newline at end of file
+        return np.array([embedding.tolist() for embedding in embeddings], dtype=np.float32)
+
+
+def get_premai_api_key(api_key: Optional[str] = None) -> str:
```
Comment (krypticmouse) [inline dsp/modules/sentence_vectorizer.py:258]: https://github.com/stanfordnlp/dspy/blob/c6219ca462baab32cddc671d978ad3190f31cec2/dsp/modules/premai.py#L38

We can use this function here by importing it.

**Review by krypticmouse — COMMENTED** [review]
Few edits but LGTM! Thanks a lot for the contribution!

File: dspy/datasets/dataloader.py
Comment (krypticmouse) [inline dspy/datasets/dataloader.py:1]: Possible to leave this file out?

**Review by krypticmouse — CHANGES_REQUESTED** [review]
Few edits but LGTM! Thanks a lot for the contribution!

File: dspy/datasets/dataloader.py
Comment (Anindyadeep) [reply dspy/datasets/dataloader.py:1]: Yes done

File: dsp/modules/sentence_vectorizer.py (hunk 1 above)
Comment (Anindyadeep) [reply dsp/modules/sentence_vectorizer.py]: Thanks for pointing it out, fixed :)

**Review by krypticmouse — APPROVED** [review]
LGTM :ship:. Thanks for the contribution!!

**Pull request merged**